MaterialLib/Fluid/Viscosity/createViscosityModel.cpp

```diff
@@ -0,0 +1,127 @@
+/*!
+   \file  createViscosityModel.cpp
+
+   \copyright
+    Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+               Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+*/
+
+#include "createViscosityModel.h"
+
+#include "BaseLib/Error.h"
+
+#include "MaterialLib/Fluid/ConstantFluidProperty.h"
+#include "LinearPressureDependentViscosity.h"
+#include "TemperatureDependentViscosity.h"
+#include "VogelsLiquidDynamicViscosity.h"
+
+namespace MaterialLib
+{
+namespace Fluid
+{
+/*!
+    \param config  ConfigTree object which contains the input data
+                   including  <type>fluid</type> and it has
+                   a tag of <viscosity>
+*/
+static std::unique_ptr<FluidProperty> createLinearPressureDependentViscosity(
+    BaseLib::ConfigTree const& config)
+{
+    std::array<double, 3> parameters = {
+        {//! \ogs_file_param{material__fluid__viscosity__temperature_dependent__mu0}
+         config.getConfigParameter<double>("mu0"),
+         //! \ogs_file_param{material__fluid__viscosity__temperature_dependent__p0}
+         config.getConfigParameter<double>("p0"),
+         //! \ogs_file_param{material__fluid__viscosity__temperature_dependent__gamma}
+         config.getConfigParameter<double>("gamma")}};
+    return std::unique_ptr<FluidProperty>(
+        new LinearPressureDependentViscosity(parameters));
+}
+
+/*!
+    \param config  ConfigTree object which contains the input data
+                   including  <type>fluid</type> and it has
+                   a tag of <viscosity>
+*/
+static std::unique_ptr<FluidProperty> createTemperatureDependentViscosity(
+    BaseLib::ConfigTree const& config)
+{
+    std::array<double, 3> parameters = {
+        {//! \ogs_file_param{material__fluid__viscosity__temperature_dependent__mu0}
+         config.getConfigParameter<double>("mu0"),
+         //! \ogs_file_param{material__fluid__viscosity__temperature_dependent__tc}
+         config.getConfigParameter<double>("tc"),
+         //! \ogs_file_param{material__fluid__viscosity__temperature_dependent__tv}
+         config.getConfigParameter<double>("tv")}};
+    return std::unique_ptr<FluidProperty>(
+        new TemperatureDependentViscosity(parameters));
+}
+
+std::unique_ptr<FluidProperty> createViscosityModel(
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{material__fluid__viscosity__type}
+    auto const type = config.getConfigParameter<std::string>("type");
+
+    if (type == "Constant")
+        return std::unique_ptr<FluidProperty>(new ConstantFluidProperty(
+            //! \ogs_file_param{material__fluid__viscosity__Constant__value}
+            config.getConfigParameter<double>("value")));
+    //! \ogs_file_param{material__fluid__viscosity__LinearPressure}
```

Comment:
I think lines 72, 75 and 78 have to moved directly above the corresponding `config.getConfigParameter<std::string>` lines.